src/main/kotlin/uk/gov/justice/digital/hmpps/educationemploymentapi/config/ControllerAdvice.kt

```diff
@@ -0,0 +1,149 @@
+package uk.gov.justice.digital.hmpps.educationemploymentapi.config
+
```

Comment:
The exceptions handled here are different to those originally handled.
The previous ExceptionHandler dealt with MethodArgumentTypeMismatchException, this does not.
If for instance the client calls POST /readiness-profiles/A1245BB/notes/DISCLOSURE_LETTERX, the error response is now:

{
    "status": 500,
    "errorCode": null,
    "userMessage": "Unexpected error: Failed to convert value of type 'java.lang.String' to required type 'uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo'; nested exception is org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.lang.String] to type [@io.swagger.v3.oas.annotations.media.Schema @org.springframework.web.bind.annotation.PathVariable uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo] for value 'DISCLOSURE_LETTERX'; nested exception is java.lang.IllegalArgumentException: No enum constant uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo.DISCLOSURE_LETTERX",
    "developerMessage": "Failed to convert value of type 'java.lang.String' to required type 'uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo'; nested exception is org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.lang.String] to type [@io.swagger.v3.oas.annotations.media.Schema @org.springframework.web.bind.annotation.PathVariable uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo] for value 'DISCLOSURE_LETTERX'; nested exception is java.lang.IllegalArgumentException: No enum constant uk.gov.justice.digital.hmpps.educationemploymentapi.data.jsonprofile.ActionTodo.DISCLOSURE_LETTERX",
    "moreInfo": null
}

Which is not true. The client made a bad request (400) - now it implies there was an internal, application error 500